go/caddy/pfipset/handlers.go

```diff
@@ -70,20 +97,29 @@ func handleIsolationPassthrough(res http.ResponseWriter, req *http.Request) {
 
 func handleLayer2(res http.ResponseWriter, req *http.Request) {
 	IPSET := pfIPSETFromContext(req.Context())
-	vars := mux.Vars(req)
-	IP := vars["ip"]
-	Mac := vars["mac"]
-	Network := vars["network"]
-	Type := vars["type"]
-	Catid := vars["category_id"]
-	Local := vars["local"]
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		panic(err)
+	}
+	var o Options
+	err = json.Unmarshal(body, &o)
+	if err != nil {
+		panic(err)
+	}	
+	IP := o.IP
```

Comment:
I think that since those options are required, we should perform validation that they are not empty since when they were in a URL we were 100% sure they would be there, while now they could be omitted by mistake